src/components/LogRow/BaseRow/SharingMenu/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { QueryParams } from "constants/queryParams";
+import { size } from "constants/tokens";
+import { useLogContext } from "context/LogContext";
+import { useMultiLineSelectContext } from "context/MultiLineSelectContext";
+import { useToastContext } from "context/toast";
+import { useQueryParams } from "hooks/useQueryParam";
+import { copyToClipboard, getJiraFormat } from "utils/string";
+import { getLinesInProcessedLogLinesFromSelectedLines } from "./utils";
+
+interface SharingMenuProps {
+  defaultOpen: boolean;
+}
+
+const SharingMenu: React.FC<SharingMenuProps> = ({ defaultOpen }) => {
+  const { selectedLines } = useMultiLineSelectContext();
+  const { getLine, processedLogLines } = useLogContext();
+  const [params, setParams] = useQueryParams();
+  const dispatchToast = useToastContext();
+  const [open, setOpen] = useState(defaultOpen);
+  const { sendEvent } = useLogWindowAnalytics();
+
+  const handleToggleMenu = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    if (open) {
+      sendEvent({ name: "Closed Share Menu" });
+    } else {
+      sendEvent({ name: "Opened Share Menu" });
+    }
+    setOpen((o) => !o);
+  };
+
+  const handleCopySelectedLines = async () => {
+    const { endingLine, startingLine } = selectedLines;
+    if (startingLine === undefined || endingLine === undefined) return;
+    // Create an array of line numbers that represent the range in selectedLines
+    const lineNumbers = getLinesInProcessedLogLinesFromSelectedLines(
+      processedLogLines,
+      selectedLines
+    );
+
+    await copyToClipboard(getJiraFormat(lineNumbers, getLine));
+    setOpen(false);
+    sendEvent({
+      name: "Copied Share Lines To Clipboard",
+    });
+    dispatchToast.success(`Copied ${lineNumbers.length} lines to clipboard`);
+  };
+
+  const handleOnlySearchOnRange = () => {
+    const { endingLine, startingLine } = selectedLines;
+    if (startingLine === undefined || endingLine === undefined) return;
+    setParams({
+      ...params,
+      [QueryParams.LowerRange]: startingLine,
+      [QueryParams.UpperRange]: endingLine,
+    });
+    setOpen(false);
+    sendEvent({
+      name: "Applied Range Limit",
+    });
+  };
+
+  const handleShareLinkToSelectedLines = async () => {
+    const { startingLine } = selectedLines;
+    if (startingLine === undefined) return;
+    // Take the current URL and add the shareLine query param
+    const url = new URL(window.location.href);
+    url.searchParams.set(QueryParams.ShareLine, startingLine.toString());
+
+    await copyToClipboard(url.toString());
+    setOpen(false);
+    sendEvent({ name: "Copied Share Link" });
+    dispatchToast.success(`Copied link to clipboard`);
+  };
+
+  const lineCount =
+    selectedLines.endingLine === undefined ||
+    selectedLines.startingLine === undefined
+      ? 1
+      : selectedLines.endingLine - selectedLines.startingLine + 1;
+
+  return (
+    <Menu
+      data-cy="sharing-menu"
+      open={open}
+      trigger={
```

Comment:
Scrolling the page leaves the menu behind, it would be nice for it to be attached to the trigger (though again this may be more of an LG issue)

Reply:
Did this end up too difficult to implement?

Reply:
It did. Since this uses a portal to render this element it required using a ref on the scrollable container and propagating it down to the component. But since the scrollable container is handled internally by react-virtuoso it wasn't as easy to expose.